src/main/java/faang/school/accountservice/controller/AccountController.java

```diff
@@ -0,0 +1,63 @@
+package faang.school.accountservice.controller;
+
+import faang.school.accountservice.dto.AccountDto;
+import faang.school.accountservice.dto.UpdateAccountDto;
+import faang.school.accountservice.service.AccountService;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("v1/accounts")
+public class AccountController {
+    private final AccountService accountService;
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public AccountDto openAccount(@RequestBody @Valid AccountDto accountDto) {
+        return accountService.openAccount(accountDto);
+    }
+
+    @PutMapping("/{accountId}/update")
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    public AccountDto updateAccount(@PathVariable long accountId, @RequestBody @Valid UpdateAccountDto updateAccountDto) {
+        return accountService.updateAccount(accountId, updateAccountDto);
+    }
+
+
+    @GetMapping("/{accountId}")
+    @ResponseStatus(HttpStatus.FOUND)
+    public AccountDto getAccount(@PathVariable long accountId) {
+        return accountService.getAccount(accountId);
+    }
+
+    @PutMapping("/{accountId}/block")
+    @ResponseStatus(HttpStatus.OK)
+    public void blockAccount(@PathVariable long accountId) {
+        accountService.blockAccount(accountId);
+    }
+
+    @PutMapping("/{accountId}/unblock")
+    @ResponseStatus(HttpStatus.OK)
+    public void unblockAccount(@PathVariable long accountId) {
+        accountService.unblockAccount(accountId);
+    }
+
+    @PutMapping("/{accountId}/close")
```

Comment:
/close/{accountId} makes more sense, it's more conventional approach.

Reply:
done